mysqlparse/grammar/alter_table.py

```diff
@@ -144,10 +144,14 @@
 # Source: http://dev.mysql.com/doc/refman/5.7/en/alter-table.html
 #
 
+_database_name = (Optional(Word(alphanums + "`_") + FollowedBy('.') +
+                           Suppress('.'), default=None)
+                  .setResultsName("database_name")
+                  .setParseAction(lambda toks: toks[0]))
```

Comment:
database and table names actually support way more characters than alphanums and _.. https://dev.mysql.com/doc/refman/5.7/en/identifiers.html

however the existing code is only supporting alphanums for table names, so this is probably a follow-up issue @seporaitis